src/core/components/ChatInput.tsx

```diff
@@ -183,56 +184,14 @@ const ChatInput = ({ setShouldAutoScroll }: ChatInputProps) => {
     return true;
   }, [imageIDs.length, maximumFileUploads, resetUploadState]);
 
-  const processFile = useCallback(
-    async (file: File) => {
-      if (!hasAvailableUploads()) {
-        return;
-      }
-
-      if (file.size > maximumFileBytes) {
-        setUploadingState({
-          isActive: true,
-          isSupportedFile: false,
-          errorMessage: 'File too large! Maximum file size is 8MB.',
-        });
-
-        // Present the error for a short time, then reset
-        setTimeout(() => {
-          resetUploadState();
-        }, 2000);
-
-        return;
-      }
-
-      if (!isSupportedFileType(file.type)) {
-        setUploadingState({
-          isActive: true,
-          isSupportedFile: false,
-          errorMessage:
-            'Invalid file type! Please upload a JPEG, PNG, or WebP image.',
-        });
-
-        //  Present the error for a short time, then reset
-        setTimeout(() => {
-          resetUploadState();
-        }, 1000);
-
-        return;
-      }
-
-      resetUploadState();
-
-      const reader = new FileReader();
-      reader.onload = async (e) => {
-        if (e.target && typeof e.target.result === 'string') {
-          const imgID = await saveImage(e.target.result);
-          setImageIDs((prevIDs) => [...prevIDs, imgID]);
-        }
-      };
-      reader.readAsDataURL(file);
-    },
-    [hasAvailableUploads, maximumFileBytes, resetUploadState],
-  );
+  const processUploadedFile = useProcessUploadedFile({
+    hasAvailableUploads,
+    maximumFileBytes,
+    setUploadingState,
+    saveImage,
+    setImageIDs,
+    resetUploadState,
+  });
```

Comment:
Eventually, `saveImage` and `hasAvailableUploads` will also be isolated, so they can be imported and won't need to be arguments